src/formats/jsonLd.ts

```diff
@@ -74,56 +48,23 @@ export const getJsonLdParser = (): Parser => {
     onComplete: (callback) => {
       onCompleteCallbacks.push(callback);
     },
-    parse: async (source, resourceInfo) => {
-      const jsonld = (await import("jsonld")).default;
+    parse: async (source, resourceInfo) =>
+      new Promise((res) => {
+        const parser = new JsonLdParser({
+          baseIRI: getSourceUrl(resourceInfo),
+          documentLoader: new FetchDocumentLoader(fetch),
+        });
+        parser.on("end", () => res());
+        parser.on("error", () => res());
```

Comment:
Would it make sense to have the parsing promise reject if there is a parsing error?

Reply:
Usually I would say yes - but in this instance it would be a breaking change because the original parser implementation using `jsonld` also resolved regardless of whether or not there was a parsing error.

Reply:
I think it would be okay to reject honestly, it feels like a bug to me that there isn't an error in case of parsing issue. What do you think?

Reply:
Well if you look at the `Parser` type that this is meant to be implementing it actually is meant to return a `void` rather than a `Promise<void>` (which is what is actually returned before and after the migration). So the expected use of this function would appear to be something like the following where the `parse` method is not even awaited

```ts
const parser = getJsonLdParser();

const quadArr = [];

await new Promise((res, rej) => {
 parser.onQuad(quadArr.push);
 parser.onError(rej);
 parser.onComplete(res)
parser.parse("My string");
});

console.log(quadArr)
```

In this case if we reject then an error is going to resolve in an uncaught reject which will end node process (in most cases unexpectedly) after 1 second then an error will be thrown

Reply:
So imagine I that this will be a breaking change to anyone who does use the parser

Reply:
But this raises another good point; we should add a test to make sure that the `onError` callbacks get triggered before the `onComplete` callbacks for cases like this.

Reply:
Oh right I hadn't realized `parse` wasn't an `async`. Yeah in that case rejecting isn't an option, any rejection should actually happen in a wrapping promise like https://github.com/inrupt/solid-client-js/blob/cf381af7eb0ca77853d90aa2ef49013986837eed/src/resource/solidDataset.ts#L205 does. And having a test to make sure things are called in the appropriate order definitely makes sense.

Reply:
Those tests have been added :)